### PR TITLE
New tests environment

### PR DESCRIPTION
**Pull request by ellisonleao**
Using the same text as #469 but now merging to the right branch.

### CODE REVIEW

File: src/system/device.js [hunk 1]
```diff
@@ -112,27 +112,30 @@
...
-                        }
-                    } else {
-                        if (me.sys.stopOnBlur) {
-                            me.state.restart(true);
-                        }
-                        if (me.sys.resumeOnFocus) {
-                            me.state.resume(true);
+                 
+            // register on the event if supported
+            if (typeof (visibilityChange) === "string") {
```
Comment (agmcleod) [inline src/system/device.js:117]: @ellisonleao What's the purpose behind the conditional change here?

Comment (obiot) [inline src/system/device.js:117]: CocoonJS compatibility, as `visibilityChange` remains undefined in that case !

(this is from master though)

Comment (agmcleod) [inline src/system/device.js:117]: Ah okay, was just surprised to see changes to the actual logic in the PR :)

Comment (ellisonleao) [inline src/system/device.js:117]: Yeah that was not me, this came with the master branch merge. @agmcleod

Comment (agmcleod) [inline src/system/device.js:117]: ah okay. Interesting that we're seeing the diff.

**Pull request merged**